compiler/optimizer/LoopVersioner.cpp

```diff
@@ -3444,7 +3444,7 @@ void TR_LoopVersioner::updateDefinitionsAndCollectProfiledExprs(TR::Node *parent
                 if ((valueInfo->getTopProbability() > MIN_PROFILED_FREQUENCY) &&
                     (valueInfo->getTotalFrequency() > 0) &&
                     !_containsCall &&
-                    !node->getByteCodeInfo().doNotProfile() && !node->isNodeCreatedByPRE() &&
+                    !node->getByteCodeInfo().doNotProfile() && !node->chkNodeCreatedByPRE() &&
```

Comment:
I think this condition can be left as `isNodeCreatedByPRE()`.  The test [above](https://github.com/eclipse/omr/blob/5bb1f0f6e05d90b8bb146cd5807b9f79cce7ad4d/compiler/optimizer/LoopVersioner.cpp#L3435-L3436) of 

```
           node->getOpCode().isLoadVar() &&
node->getSymbolReference()->getSymbol()->isAutoOrParm()))
```

should mean that the opcode will satisfy `isLoadVarDirect`.

If I understand things correctly, the various `chk<flag>` methods are used exclusively by Debug.cpp for printing the IL, while the `is<flag>` methods are used elsewhere for testing conditions.  It's those `is<flag>` method calls where we really want the JIT compiler to check assertions to ensure it's not working with some operation in a context where it wasn't expected.